typify-impl/src/merge.rs

```diff
@@ -952,11 +1023,22 @@ fn merge_so_object(
                         Ok(schema) => Some(Ok((name.clone(), schema))),
                     }
                 })
-                .collect::<Result<_, _>>()?;
+                .collect::<Result<schemars::Map<_, _>, _>>()?;
 
             let additional_properties = additional_properties.map(Box::new);
             let required = aa.required.union(&bb.required).cloned().collect();
 
+            // So. We merged the properties and we merged the array of
+            // properties that are required. We use the absence of a property
+            // in the properties map to indicate that the property must *not*
+            // be present. This is imprecise, but allows us to make progress
+            // without a most significant conversion to a new representation.
```

Comment:
we could change this to see if the property was invalid i.e. `Schema::Bool(false)`